Sources/MongoSwift/MongoClient.swift

```diff
@@ -278,6 +278,96 @@ public class MongoClient {
+      *   - `ServerError.commandError` if an error occurs on the server while creating the cursor.
+      *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination or the pipeline passed
+      *     is invalid.
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      * - SeeAlso: https://docs.mongodb.com/manual/reference/system-collections/
+      */
+     public func watch<T: Codable>(_  pipeline: [Document],
+                                   options: ChangeStreamOptions?  =  nil,
+                                   session: ClientSession? = nil,
+                                   withFullDocumentType: T.Type) throws ->
+                                   ChangeStream<ChangeStreamDocument<T>> {
+        let pipeline: Document = ["pipeline": pipeline]
+        let connection = try self.connectionPool.checkOut()
+        let opts = try encodeOptions(options: options, session: session)
+        let changeStreamPtr: OpaquePointer = mongoc_client_watch(self._client, pipeline._bson, opts?._bson)
+        return try ChangeStream<ChangeStreamDocument<T>>(stealing: changeStreamPtr,
+                                                         client: self,
+                                                         connection: connection,
+                                                         session: session,
+                                                         decoder: self.decoder)
+     }
+
+     /**
+      * Starts a `ChangeStream` on a `MongoClient`. Allows the client to observe all changes in a cluster - excluding
+      * system collections i.g. "config", "local", and "admin" databases. Associates the specified `Codable` type `T`
+      * with the returned `ChangeStream`.
+      * - Parameters:
+      *   - Pipeline: The pipeline of stages to append to an initial `ChangeStream` stage.
+      *   - Options: An optional `ChangeStreamOptions` to use on the initial `ChangeStream` stage.
+      *   - Session: An optional `ChangeStream` to use with this change stream.
+      * - Returns: A `ChangeStream` on all collections in all databases in a cluster.
+      * - Throws:
+      *   - `ServerError.commandError` if an error occurs on the server while creating the cursor.
+      *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination or the pipeline passed
+      *     is invalid.
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      * - SeeAlso: https://docs.mongodb.com/manual/reference/system-collections/
+      */
+     public func watch<T: Codable>(_  pipeline: [Document],
+                                   options: ChangeStreamOptions?  =  nil,
+                                   session: ClientSession? = nil,
+                                   withFullDocumentType: T.Type) throws ->
```

Comment:
The `T` here isn't the type of the `fullDocument` field--it's the type of the entire change stream document itself. This parameter label should instead be something like `withReturnType` (as per the design doc) or `withEventType`. I kind of prefer one of the latter two since this type affects what the change stream returns, not what this method returns.

Reply:
Oops, yes, the parameter it was supposed to be labeled `withReturnType`.

Reply:
@kmahar @mbroadst Any opinions on `withEventType` vs `withReturnType` for the parameter label?

Reply:
I like `withEventType`

Reply:
I'm fine with that, though I wonder if deviating from the spec there is problematic

Reply:
Another option (which is what rust is doing) is `ChangeStreamEventDocument`, which deviates slightly less from the spec but is pretty verbose.

Reply:
hmm... ya i think i'd maybe just go with `ChangeStreamEvent` over that